src/github_runner_image_builder/builder.py

```diff
@@ -0,0 +1,635 @@
+                timeout=60 * 10,
+            )
+            logger.info("git clone out: %s", output)
+        else:
+            output = subprocess.check_output(  # nosec: B603
+                ["/usr/bin/git", "-C", str(YQ_REPOSITORY_PATH), "pull"],
+                timeout=60 * 10,
+            )
+            logger.info("git pull out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/snap/bin/go", "build", "-C", str(YQ_REPOSITORY_PATH), "-o", str(HOST_YQ_BIN_PATH)],
+            timeout=20 * 60,
+        )
+        logger.info("go build out: %s", output)
+        shutil.copy(HOST_YQ_BIN_PATH, MOUNTED_YQ_BIN_PATH)
+    except subprocess.CalledProcessError as exc:
+        logger.exception(
+            "Error installing yq, cmd: %s, code: %s, err: %s",
+            exc.cmd,
+            exc.returncode,
+            exc.output,
+        )
+        raise YQBuildError from exc
+
+
+def _disable_unattended_upgrades() -> None:
+    """Disable unatteneded upgrades to prevent apt locks.
+
+    Raises:
+        UnattendedUpgradeDisableError: If there was an error disabling unattended upgrade related
+            services.
+    """
+    try:
+        # use subprocess run rather than operator-libs-linux's systemd library since the library
+        # does not provide full features like mask.
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "stop", APT_TIMER], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl stop apt timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "disable", APT_TIMER], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl disable apt timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "mask", APT_SVC], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl mask apt timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "stop", APT_UPGRADE_TIMER], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl stop apt upgrade timer out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/bin/systemctl", "disable", APT_UPGRADE_TIMER], timeout=30
+        )
+        logger.info("systemctl disable apt upgrade timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "mask", APT_UPGRAD_SVC], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl mask apt upgrade timer out: %s", output)
+        output = subprocess.check_output(
+            ["/usr/bin/systemctl", "daemon-reload"], timeout=30
+        )  # nosec: B603
+        logger.info("systemctl daemon-reload out: %s", output)
+        output = subprocess.check_output(  # nosec: B603
+            ["/usr/bin/apt-get", "remove", "-y", "unattended-upgrades"], timeout=30
```

Comment:
Shouldn't `DEBIAN_FRONTEND` be set here ?

Reply:
Set, thank you!